File: web/src/types/system.ts
export interface SystemInfo {
  id: number
  config_key: string
  config_value: string
  config_type: string
  description: string
  status: number
  created_at: string
  updated_at: string
  created_by: number
  updated_by: number
}

export interface SystemInfoList {
  id: number
  config_key: string
  config_value: string
  config_type: string
  description: string
  status: number
}

// 根据后端实际提供的配置项定义
export interface SystemConfig {
  system_name?: string
  system_icon?: string
  system_logo?: string
  system_description?: string
  [key: string]: string | undefined
}

File: server/models/system.go
package models

import "time"

type System struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	ConfigKey   string    `gorm:"not null;size:100" json:"config_key"`
	ConfigValue string    `gorm:"not null" json:"config_value"`
	ConfigType  string    `gorm:"not null;size:50" json:"config_type"`
	Description string    `gorm:"size:255" json:"description"`
	Status      int       `gorm:"default:1" json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	CreatedBy   uint      `json:"created_by"`
	UpdatedBy   uint      `json:"updated_by"`
}

func (System) TableName() string {
	return "system"
}
